pkg/resources/grant_ownership_acceptance_test.go

```diff
@@ -317,6 +317,146 @@ func TestAcc_GrantOwnership_OnObject_Table_ToDatabaseRole(t *testing.T) {
 	})
 }
 
+func TestAcc_GrantOwnership_OnObject_Procedure_ToAccountRole(t *testing.T) {
+	databaseId := acc.TestClient().Ids.RandomAccountObjectIdentifier()
+	databaseName := databaseId.Name()
+	schemaName := acc.TestClient().Ids.Alpha()
+	withArgProcedureName := sdk.NewSchemaObjectIdentifierWithArguments(databaseName, schemaName, acc.TestClient().Ids.Alpha(), []sdk.DataType{sdk.DataTypeVARCHAR})
+	withoutArgProcedureName := sdk.NewSchemaObjectIdentifierWithArguments(databaseName, schemaName, acc.TestClient().Ids.Alpha(), []sdk.DataType{})
+
+	accountRoleId := acc.TestClient().Ids.RandomAccountObjectIdentifier()
+	accountRoleName := accountRoleId.Name()
+	accountRoleFullyQualifiedName := accountRoleId.FullyQualifiedName()
+
+	configVariables := config.Variables{
+		"account_role_name":          config.StringVariable(accountRoleName),
+		"database_name":              config.StringVariable(databaseName),
+		"schema_name":                config.StringVariable(schemaName),
+		"with_arg_procedure_name":    config.StringVariable(withArgProcedureName.Name()),
+		"without_arg_procedure_name": config.StringVariable(withoutArgProcedureName.Name()),
+	}
+	withArgResourceName := "snowflake_grant_ownership.with_arguments"
+	withoutArgResourceName := "snowflake_grant_ownership.without_arguments"
+
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+		PreCheck:                 func() { acc.TestAccPreCheck(t) },
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
+		Steps: []resource.TestStep{
+			{
+				ConfigDirectory: acc.ConfigurationDirectory("TestAcc_GrantOwnership/OnObject_Procedure_ToAccountRole"),
+				ConfigVariables: configVariables,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(withArgResourceName, "account_role_name", accountRoleName),
+					resource.TestCheckResourceAttr(withArgResourceName, "on.0.object_type", "PROCEDURE"),
+					resource.TestCheckResourceAttr(withArgResourceName, "on.0.object_name", withArgProcedureName.FullyQualifiedName()),
+					resource.TestCheckResourceAttr(withArgResourceName, "id", fmt.Sprintf("ToAccountRole|%s||OnObject|PROCEDURE|%s", accountRoleFullyQualifiedName, withArgProcedureName.FullyQualifiedName())),
+					checkResourceOwnershipIsGranted(&sdk.ShowGrantOptions{
+						To: &sdk.ShowGrantsTo{
+							Role: accountRoleId,
+						},
+					}, sdk.ObjectTypeProcedure, accountRoleName, fmt.Sprintf("%s.%s.\"%s(ARG1 VARCHAR):VARCHAR(16777216)", databaseName, schemaName, withArgProcedureName.Name())),
```

Comment:
The last character should end with "\", but it is trimmed somewhere, so it is written this way.
It would be better to modify it so that it can be asserted with the correct value, but I did not modify it because I did not want to expand the scope as much as possible.